src/current/advisories/a131639.md

```diff
@@ -0,0 +1,211 @@
+
+All CockroachDB versions prior to and including the following are impacted by this bug:
+
+- v23.1.0 - v23.1.26 and below
+- v23.2.0 - v23.2.10 and below
+- v24.1.0 - subsequent patch versions are unaffected
+
+Versions prior to 23.1 are no longer eligible for [maintenance support]({% link releases/release-support-policy.md %}), and this issue will not be addressed in those versions.
+
+## Statement
+
+[#123442](https://github.com/cockroachdb/cockroach/pull/123442) resolves an issue in CockroachDB in the expiration-to-epoch lease promotion transition, where the lease's effective expiration could be allowed to regress, resulting in two nodes believing they are the leaseholder for a range.
+
+The patch has been applied to maintenance releases of CockroachDB:
+
+- [v23.1.27]({% link releases/v23.1.md%}#v23-1-27)
+- [v23.2.11]({% link releases/v23.2.md%}#v23-2-11)
+- [v24.1.1]({% link releases/v24.1.md%}#v24-1-1)
+
+This public issue is tracked by [131639](https://github.com/cockroachdb/cockroach/issues/131639).
+
+## Mitigation
+
+Users are encouraged to upgrade to [v23.1.27]({% link releases/v23.1.md%}#v23-1-27), [v23.2.11]({% link releases/v23.2.md%}#v23-2-11), [v24.1.1]({% link releases/v24.1.md%}#v24-1-1), or a later version that includes the patch.
+
+### Detection via logs
+
+Users can run the script included below to detect part of the symptoms of this technical advisory. The script accepts a [debug.zip]({% link v24.3/cockroach-debug-zip.md %}) and determines whether any lease expiration regressions are evident from the logs.
+
+As described previously, a lease expiration regression is only one of the two race conditions needed to encounter a lost write, with the other being a stale read during asynchronous intent resolution on the range that had the lease expiration regression. Consequently, this script does not provide a complete method of detection for a lost write. Running this script may result in a limited number of false positives. For complete confirmation, this script should be combined with the method described in [Detection via an audit of primary and secondary indexes](#detection-via-an-audit-of-primary-and-secondary-indexes).
+
+Users should not run the script on a production system. Instead, users should move the `debug.zip` file elsewhere for analysis and run the script there.
+
+The script requires (and asks for) the following dependencies:
+
+- [ripgrep](https://github.com/BurntSushi/ripgrep#installation)
+- [DuckDB](https://duckdb.org/docs/installation)
+
+Example usage:
+
+~~~
+➜ ./ta131639_check.sh debug.prod-crdb2.20240922001746.zip
```

Comment:
This output is a little stale and also includes more information than we should probably share externally. How about:
```bash
➜ pwd
/path/of/unzipped/debug

➜ /path/to/script/ta131639_check.sh
Searching for liveness epoch increments ...
Searching for slow lease promotions ...
Querying the logs for lease expiration regressions ...

❌ Lease expiration regressions found. Symptoms detected.

symptoms
node 123 observed possible symptoms at 2024-09-01 12:34:56 on ranges 123456
```